src/protocol-handler-installer.js

```diff
@@ -25,32 +21,32 @@ class ProtocolHandlerInstaller {
     return this.isSupported() && remote.app.setAsDefaultProtocolClient('atom', process.execPath, ['--uri-handler', '--'])
   }
 
-  shouldUpgradeProtocolClient () {
-    // macOS and Linux ignore the last argument to `app.isDefaultProtocolClient`
-    // so we only need to upgrade the handler on Windows.
-    return process.platform === 'win32' && this.isOldDefaultProtocolClient()
-  }
-
   initialize (config, notifications) {
     if (!this.isSupported()) {
       return
     }
 
-    if (this.shouldUpgradeProtocolClient()) {
-      this.setAsDefaultProtocolClient()
-    } else if (!this.isDefaultProtocolClient()) {
-      const behaviorWhenNotProtocolClient = config.get(SETTING)
-      switch (behaviorWhenNotProtocolClient) {
-        case PROMPT:
+    const behaviorWhenNotProtocolClient = config.get(SETTING)
+    switch (behaviorWhenNotProtocolClient) {
+      case PROMPT:
+        if (!this.isDefaultProtocolClient()) {
           this.promptToBecomeProtocolClient(config, notifications)
-          break
-        case ALWAYS:
+        }
+        break
+      case ALWAYS:
+        if (!this.isDefaultProtocolClient()) {
           this.setAsDefaultProtocolClient()
-          break
-        case NEVER:
-        default:
-          // Do nothing
-      }
+        }
+        break
+      case NEVER:
+        if (process.platform === 'win32') {
+          // Only win32 supports deregistration
+          const Registry = require('winreg')
+          const commandKey = new Registry({hive: 'HKCR', key: `\\atom`})
```

Comment:
Will the protocol client definitely be the only thing in this location?

Reply:
Yeah, all our other registry things live under HKCU (current user).  You can see them in WinShell.js